app/controllers/step_by_step_pages_controller.rb

```diff
@@ -165,16 +166,11 @@ def revert_page
   def generate_internal_change_note(intent)
     change_note = @step_by_step_page.internal_change_notes.new(
       author: current_user.name,
-      description: note_description(intent)
+      description: intent.note_description(current_user)
```

Comment:
You shouldn't need to pass in `current_user`. It's a special variable added in one of the gems that accessible from the controllers.

Reply:
We've just tried this but it made it difficult to test. General [advice seems to be](https://stackoverflow.com/questions/22076699/rspec-model-testing-current-user-in-model) to pass `current_user` from the controller into your model; the model shouldn't reference the controller.

Reply:
Ah, sorry. I think when I looked at this, it looked like `note_description` was still in the controller.